test/unit/select/index.test.tsx

```diff
@@ -53,6 +53,12 @@ test('state.value updates when props.value changes', () => {
   assert.equal(wrapper.state().value, updatedValue);
 });
 
+test('state.disabled updates when props.disabled changes', () => {
```

Comment:
Your test should test your conditional. Please test that foundation.setDisabled() is called.